### PR TITLE
Blog post about Kubernetes Response Engine with Flux v2

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>
Co-authored-by: Radhika Puthiyetath <radhika.dwaraka@gmail.com>

**What type of PR is this?**

/kind content

**Any specific area of the project related to this PR?**

/area blog

**What this PR does / why we need it**:
This PR will explain how we can use Flux v2 to set up KRE (Kubernetes Response Engine).

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+```shell
+kubectl create namespace falco
+kubectl create secret generic github-secret \
+         --from-literal GITHUB_TOKEN=$GITHUB_TOKEN \
+         --namespace falco
+````
+
+* You might notice that we are using some URLs (--notify-url) within the flags of the project. Let me explain that why we need it a bit. As you might know, we are trying to set up some remediation engine here, so we have to react to those events thrown by the Falcosidekick as soon as possible. We defined an interval while creating a `GitRepository`, which means that Flux v2 will wait at least that long to sync configurations, so we have to notify Flux controllers about changes once we edited the desired state by a process whose name is `falco-event-listener.` To notify the Flux controllers about changes in Git or Helm repositories, we can set up webhooks and trigger a cluster reconciliation every time a source changes. For more detail, please [see](https://fluxcd.io/docs/guides/webhook-receivers/).
+
+There are different kind of webhook receivers in Flux v2 but we'll use the `generic` one in this guide.
```
Comment (leogr) [inline content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md:157]:
```suggestion
There are different kinds of webhook receivers in Flux v2, but we'll use the `generic` one in this guide.
```

File: content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+- cpus: 3
+- driver: virtualbox
+- memory: 8192
+
+minikube start
+```
+
+### Install Flux V2 - GitOps Toolkit
+
+I highly recommended that you check out [getting started](https://fluxcd.io/docs/get-started/) page of `Flux v2`. It gives you detailed installation instructions for Flux v2.
```
Comment (leogr) [inline content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md:71]:
```suggestion
I highly recommended that you check out the [getting started](https://fluxcd.io/docs/get-started/) page of `Flux v2`. It gives you detailed installation instructions for Flux v2.
```

File: content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+    --source alpine \
+    --path "./" \
+    --prune true \
+    --validation client \
+    --interval 5m
+```
+
+### Install falco-event-listener
+
+As the name suggests, this program will listen to an event in the form of `CloudEvents.` This CloudEvents forwarded from Falcosidekick, a simple daemon for enhancing available outputs for Falco. After the successful receipt of the event, `falco-event-listener` will update the YAML definition to scale its replicas to zero based on the pieces of information given in the event.
```
Comment (leogr) [inline content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md:127]:
```suggestion
As the name suggests, this program will listen to an event in the form of `CloudEvents`. This CloudEvents forwarded from Falcosidekick, a simple daemon for enhancing available outputs for Falco. After the successful receipt of the event, `falco-event-listener` will update the YAML definition to scale its replicas to zero based on the pieces of information given in the event.
```

File: content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+
+In most basic form, the architecture of the demo is:
+
+Falco w/HTTP --> Falcosidekick w/CloudEvent --> falco-event-listener w/HTTP --> GitHub
+
+To learn more about `CloudEvents`, see [cloudevents.io](https://cloudevents.io).
+
+First, let us clone the falco-event-listener repository.
```
Comment (leogr) [inline content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md:135]:
```suggestion
First, let us clone the `falco-event-listener` repository.
```

**Review by leogr — COMMENTED** [review]
Nice blog, looking good! :clap:  I just found a few minor issues.

File: content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md (hunk 4 above)
Comment (developer-guy) [reply content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md]: thank you so much for fixing the issues, it should be fixed now 🙏

File: content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md (hunk 3 above)
Comment (developer-guy) [reply content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md]: thank you so much for fixing the issues, it should be fixed now 🙏

File: content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md (hunk 2 above)
Comment (developer-guy) [reply content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md:71]: thank you so much for fixing the issues, it should be fixed now 🙏

File: content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md (hunk 1 above)
Comment (developer-guy) [reply content/en/blog/falcosidekick-response-engine-part-8-fluxv2.md]: thank you so much for fixing the issues, it should be fixed now 🙏

**Comment by developer-guy** [comment]
@leogr just a quick note about the process here, it would be better to get feedback from @stealthybox just before merging it because AFAIK he is also working on this blog post too. 🙏

**Comment by squaremo** [comment]
Quick note: please can you refer to "Flux v2" only initially (perhaps spelling it out as "Flux (version 2)"), and thereafter just "Flux". This makes the text flow a little better -- and avoids implying there is another Flux that people should be aware of. Thank you!

**Comment by developer-guy** [comment]
> "Flux (version 2)"

thanks for the feedback @squaremo, it should be done right now, I replaced all the Flux mentions as `Flux (version 2)`.

**Review by leogr — APPROVED** [review]
LGTM!

Thank you :pray:

**Pull request merged**